### PR TITLE
login(feat): create inputValidator Dependency

### PR DESCRIPTION
**Pull request by meunomeecris**
## Description, motivation and context
- create a dependency for the func 'inputValidator'
- add unit tests for the dependency
- 
## Screenshots and GIFs (Artifacts)
Screenshot 2024-05-16 at 10 01 59
Screenshot 2024-05-16 at 10 02 12

Thank you!

### CODE REVIEW

File: mobile/evidence/ios/Features/Tests/LoginTests/InputValidatorTests.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+//
+//  Created by Cris Messias on 16/05/24.
+//
+
+import XCTest
+@testable import Login
+@testable import Models
+import Dependencies
+
+final class InputValidatorTests: XCTestCase {
```
Comment (viniciusaro) [inline mobile/evidence/ios/Features/Tests/LoginTests/InputValidatorTests.swift:13]: Maybe you could add more tests for edge cases such as `@example.com`, `test@`, `a@b`, `a@b.com.br.co.de`

**Review by viniciusaro — APPROVED** [review]
Great, my fried! :DDD

File: mobile/evidence/ios/Features/Tests/LoginTests/InputValidatorTests.swift (hunk 1 above)
Comment (meunomeecris) [reply mobile/evidence/ios/Features/Tests/LoginTests/InputValidatorTests.swift:13]: Nice idea. The test failed in this last example...
